README.md

```diff
@@ -268,7 +268,24 @@ module.exports = {
     const cwd = process.cwd()
     const relativePaths = absolutePaths.map((file) => path.relative(cwd, file))
     return `ng lint myProjectName --files ${relativePaths.join(' ')}`
-  }
+  },
+}
+```
+
+## About the shell option
+
+By default lint-staged runs the configured task using [execa](https://github.com/sindresorhus/execa#shell), which handles parsing and escaping of the command and its arguments. Lint-staged also supports using execa's `shell` option via the `--shell` flag. This skips parsing and runs the command directly in a shell, `/bin/sh` or `cmd.exe` depending on the current platform.
+
+**Using the shell option is not adviced because it's slower and insecure, and might lead to problems when running on multiple platforms!**
+
+Please note that if you are using the `shell` option with a functional configuration, _lint-staged_ cannot escape the filenames since the function is supposed to generate the entire command string. The default `escapeArg` method can be required for use:
+
+```js
+// .lintstagedrc.js
+const { escapeArg } = require('lint-staged')
+
+module.exports = {
+  '**/*.js?(x)': (filenames) => `eslint ${filenames.map(escapeArg).join(' ')}`,
 }
 ```
```

Comment:
Here we can see how to manually use the escape method in a functional task. Shouldn't be necessary when not using the `shell` option, because then the string is parsed separately and run via spawn instead of a shell.

Reply:
Maybe the method would be nicer to name just `escape` instead of `escapeArg`.